plugins/ui/src/js/src/widget/WidgetUtils.test.tsx

```diff
@@ -246,3 +247,94 @@ describe('wrapCallable', () => {
     ).rejects.toBeInstanceOf(Error);
   });
 });
+
+describe('decodeNode', () => {
+  it.each([null, undefined, '', 'test', 0, true])(
+    'should handle primitive values: %s',
+    value => {
+      const callback = jest.fn((k, v) => v);
+      expect(decodeNode(value, callback)).toBe(value);
+      expect(callback).toHaveBeenCalledWith('', value);
+    }
+  );
+
+  describe('arrays', () => {
+    it('returns original array if no children changed', () => {
+      const callback = jest.fn((k, v) => v);
+      const value = ['test', 'test2'];
+      expect(decodeNode(value, callback)).toBe(value);
```

Comment:
I actually thought another test had an unneeded check for `toBe` and `toEqual([])`, but actually points out this should be checking the values to make sure it didn't just mutate the array

```suggestion
      const value = ['test', 'test2'];
      const valueCopy = [...value];
      const result = decodeNode(value, callback);
      expect(result).toBe(value);
      expect(result).toEqual(valueCopy);
```